### ISSUE TITLE
Running evaluation for YOLOv8TFLite model

### ISSUE THREAD

**Issue by postman31** [ultralytics/ultralytics#20261]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

in https://github.com/ultralytics/ultralytics/tree/f5e15bccdde8af7c4a41e3d00e9443d17432bae9/examples/YOLOv8-TFLite-Python we have a nice example of running inference on a model converted to edge_tpu.
is there any easy way to evaluate this model over some test/validation dataset besided haveing an inference of only single image?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @postman31, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help guide you while an Ultralytics engineer reviews your question and will assist further soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for detailed guides, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions and workflows are covered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug further.

If your question is about custom training or evaluation ❓, please include as much context as possible, such as dataset examples and logs, and refer to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community for more support:
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions: [Discourse](https://community.ultralytics.com/)
- Community threads: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Ensure you're using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your question! An Ultralytics engineer will assist you soon 😊

**Comment by Y-T-G** [comment]
You can load a TFLite model in Ultralytics normally and run validation, just like the PyTorch model.

**Comment by postman31** [comment]
thanks, it works!

**Comment by glenn-jocher** [comment]
Great to hear it worked! If you need to compare the TFLite model metrics with your original PyTorch model, you can use the same validation workflow with both models:

```python
from ultralytics import YOLO

# Validate PyTorch model
pt_model = YOLO("yolo11n.pt")
pt_results = pt_model.val(data="your_dataset.yaml")

# Validate TFLite model
tflite_model = YOLO("yolo11n_edgetpu.tflite")
tflite_results = tflite_model.val(data="your_dataset.yaml")

# Compare metrics
print(f"PyTorch mAP: {pt_results.box.map}")
print(f"TFLite mAP: {tflite_results.box.map}")
```

Let us know if you have any other questions about validating your exported models.

**Comment by postman31** [comment]
It didn't work out of the box, however, I've managed to hack validator.py a little bit, forcing it to set validator device with 
`            if not(isinstance(self.device, torch.device)): ## line 186
                self.device=select_device(self.device)`
.., and passing imgsz directly to val call as 
`edgetpu_model.val(data="val_ds.yaml", imgsz=[256, 256],)`
also, autobackend seems to fail pick up tflite metadata for ftlite files
Hope I'll have some time next week to collect this as a PR

**Comment by glenn-jocher** [comment]
Thanks for sharing your workaround! Those are definitely issues that need attention - especially the metadata extraction and device handling for TFLite models. A PR with your fixes would be very welcome. The AutoBackend class (which handles model loading) should ideally handle TFLite metadata properly without requiring manual intervention. Your contribution would help improve the validation experience for everyone using exported TFLite models.
